### PR TITLE
KEYCLOAK-12862 Update hostname configuration

### PR DESCRIPTION
**Pull request by hmlnarik**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: modules/sso/config/launch/setup/74/added/launch/keycloak-spi.sh [hunk 1]
```diff
@@ -37,12 +36,24 @@ function add_vault() {
...
-# KEYCLOAK-8129 Set Keycloak server's hostname to 'request' by default if SSO_HOSTNAME not set
-function set_server_hostname_spi_to_request() {
+# KEYCLOAK-8129 Set Keycloak server's hostname to 'default' by default if SSO_HOSTNAME not set
+function set_server_hostname_spi_to_default() {
 
   if [ -z "${SSO_HOSTNAME}" ]; then
 
-    local -r hostname_spi="<spi name=\"hostname\"><default-provider>request</default-provider><provider name=\"fixed\" enabled=\"true\"><properties><property name=\"hostname\" value=\"localhost\"/><property name=\"httpPort\" value=\"-1\"/><property name=\"httpsPort\" value=\"-1\"/></properties></provider></spi>"
+    local frontend_url=''
+    if [ -n "${SSO_HOSTNAME_FRONTEND_URL:-$KEYCLOAK_FRONTEND_URL}" ]; then
```
Comment (stianst) [inline modules/sso/config/launch/setup/74/added/launch/keycloak-spi.sh]: Naming of environment variables are weird. I'd just go with SSO_FRONTEND_URL.

File: modules/sso/config/launch/setup/74/added/launch/keycloak-spi.sh [hunk 2]
```diff
@@ -37,12 +36,24 @@ function add_vault() {
...
+        frontend_url="<property name=\"frontendUrl\" value=\"${SSO_HOSTNAME_FRONTEND_URL:-$KEYCLOAK_FRONTEND_URL}\"/>"
+    fi
+    local admin_url=''
```
Comment (stianst) [inline modules/sso/config/launch/setup/74/added/launch/keycloak-spi.sh]: Makes no sense to be able to set adminUrl here, as it should then also prevent exposing the admin-url on OpenShift. So I would just remove this.

File: modules/sso/config/launch/setup/74/added/launch/keycloak-spi.sh [hunk 3]
```diff
@@ -37,12 +36,24 @@ function add_vault() {
...
+    if [ -n "${SSO_HOSTNAME_ADMIN_URL:-$KEYCLOAK_ADMIN_URL}" ]; then
+        admin_url="<property name=\"adminUrl\" value=\"${SSO_HOSTNAME_ADMIN_URL:-$KEYCLOAK_ADMIN_URL}\"/>"
+    fi
+    local force_backend_url_to_frontend_url=''
+    if [ -n "${SSO_HOSTNAME_FORCE_BACKEND_URL_TO_FRONTENT_URL:-$KEYCLOAK_FORCE_BACKEND_URL_TO_FRONTENT_URL}" ]; then
```
Comment (stianst) [inline modules/sso/config/launch/setup/74/added/launch/keycloak-spi.sh]: SSO_HOSTNAME_FORCE_BACKEND_URL_TO_FRONTENT_URL pretty crazy variable name. Not sure this is really needed as it's more a workaround in certain cases when proxy isn't configured properly. So I would leave this out.

**Comment by hmlnarik** [comment]
Thanks for the review!
Docs PR: https://github.com/keycloak/keycloak-documentation/pull/902

**Review by stianst — APPROVED** [review]

**Pull request merged**